src/error.rs

```diff
@@ -1,85 +1,170 @@
 use std::fmt;
 
-use reqwest::Method;
+use reqwest::{Method, Response};
+use serde::{Deserialize, Serialize};
+
+const GATAWAY_DELETED: &str = "gateway-deleted";
+const INVALID_CREDENTIALS: &str = "invalid-credentials";
 
 #[derive(Debug)]
```

Comment:
@silvestrpredko @jplatte, I have an idea. We can use `strum` to make deserialization less error-prone.

```suggestion
use strum::AsRefStr;

#[derive(AsRefStr, Debug)]
#[strum(serialize_all = "kebab-case")]
```

And then inside `impl<'de> Deserialize<'de> for ApiError` we can write:

```rust
match resp.code.to_kebab_case().as_str() {
    e if e == ApiError::GatewayDeleted.as_ref() => Ok(ApiError::GatewayDeleted),
    e if e == ApiError::InvalidCredentials.as_ref() => Ok(ApiError::InvalidCredentials),
    ...
```

This way we can remove string constants at all.

Isn't it awesome? :smile:

Reply:
If we're going to use `strum` we can also use `strum::EnumString` to generate half the `Deserialize` impl, using `#[strum(disabled)]` on the `Other` variant and omitting these constants.

Reply:
That is:

```rust
let resp = ApiServerResponse::deserialize(deserializer)?;
match resp.code.parse() {
    Ok(e) => e,
    Err(_) => ApiError::Other { code: resp.code, message: resp.message },
}

Reply:
Yeah, even better.

> If we're going to use strum

We already use it: https://github.com/lumeohq/api-client/blob/7f25d2ece4acd702abd38bdae516207f18ccf87d/Cargo.toml#L19